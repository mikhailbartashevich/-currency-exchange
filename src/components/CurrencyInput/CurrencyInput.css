.CurrencyInput {
  display: block;
}

.CurrencyInput_input {
  margin-right: 10px;
}
