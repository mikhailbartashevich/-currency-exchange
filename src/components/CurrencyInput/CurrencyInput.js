import React from 'react';
import './CurrencyInput.css';
import {
  Select,
  MenuItem,
  TextField,
  OutlinedInput,
  InputAdornment,
} from '@material-ui/core';
import { pure } from 'recompose';
import PropTypes from 'prop-types';
import { CurrencyPropType } from '../../model/currency.model';

const truncate = value => {
  const match = value.toString().match(/^\d+(?:\.\d{0,2})?/);
  return match ? match[0] : '';
};

const handleAmountChange = props => amount =>
  props.actions.changeAmount(truncate(amount));

const handleCurrencyChange = props => currency =>
  props.actions.changeCurrency(
    props.currencyOptions.find(option => option.currency === currency),
  );

const CurrencyInputRaw = props => {
  const currencies = props.currencyOptions || [];
  const options = currencies.map(option => (
    <MenuItem key={option.currency} value={option.currency}>
      {option.currency}
    </MenuItem>
  ));
  return (
    <div className="CurrencyInput">
      <Select
        input={<OutlinedInput />}
        className="CurrencyInput_select"
        value={props.currency ? props.currency.currency : ''}
        onChange={e => handleCurrencyChange(props)(e.target.value)}
      >
        {options}
      </Select>
      <TextField
        type="number"
        variant="outlined"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              {props.currency.symbol}
            </InputAdornment>
          ),
        }}
        value={props.amount}
        onChange={e => handleAmountChange(props)(e.target.value)}
        margin="normal"
      />
    </div>
  );
};

CurrencyInputRaw.propTypes = {
  actions: PropTypes.shape({
    changeAmount: PropTypes.func,
    changeCurrency: PropTypes.func,
  }),
  amount: PropTypes.string,
  currencyOptions: PropTypes.arrayOf(CurrencyPropType),
  currency: CurrencyPropType,
};

export const CurrencyInput = pure(CurrencyInputRaw);
