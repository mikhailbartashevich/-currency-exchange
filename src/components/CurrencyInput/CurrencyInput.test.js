import React from 'react';
import { shallow } from 'enzyme';
import { CurrencyInput } from './CurrencyInput';

describe('Balance', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CurrencyInput debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<CurrencyInput />);
    expect(component).toMatchSnapshot();
  });

  it('should render balance with given props', () => {
    const inputCurrency = { currency: 'USD', symbol: '$' };
    const changeInputAmount = () => console.log('changeInputAmount');
    const changeInputCurrency = () => console.log('changeInputCurrency');
    const currencyOptions = [
      { currency: 'USD', symbol: '$' },
      { currency: 'EUR', symbol: '€' },
      { currency: 'GBP', symbol: '£' },
    ];
    const component = shallow(
      <CurrencyInput
        amount="30"
        currency={inputCurrency}
        currencyOptions={currencyOptions}
        actions={{
          changeAmount: changeInputAmount,
          changeCurrency: changeInputCurrency,
        }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
