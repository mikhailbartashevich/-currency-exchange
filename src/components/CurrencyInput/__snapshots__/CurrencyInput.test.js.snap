// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance should render balance with given props 1`] = `
<div
  className="CurrencyInput"
>
  <WithStyles(WithFormControlContext(ForwardRef(Select)))
    onChange={[Function]}
    value="USD"
  >
    <WithStyles(ForwardRef(MenuItem))
      key="USD"
      value="USD"
    >
      USD
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="EUR"
      value="EUR"
    >
      EUR
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="GBP"
      value="GBP"
    >
      GBP
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(WithFormControlContext(ForwardRef(Select)))>
  <WithStyles(ForwardRef(TextField))
    margin="normal"
    onChange={[Function]}
    type="number"
    value="30"
  />
</div>
`;

exports[`Balance should render correctly in "debug" mode 1`] = `
<div
  className="CurrencyInput"
>
  <WithStyles(WithFormControlContext(ForwardRef(Select)))
    onChange={[Function]}
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    margin="normal"
    onChange={[Function]}
    type="number"
  />
</div>
`;

exports[`Balance should render correctly with no props 1`] = `
<div
  className="CurrencyInput"
>
  <WithStyles(WithFormControlContext(ForwardRef(Select)))
    onChange={[Function]}
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    margin="normal"
    onChange={[Function]}
    type="number"
  />
</div>
`;
